libraries/AP_Scheduler/AP_Scheduler.h

```diff
@@ -12,66 +12,60 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-/*
- *  main loop scheduler for APM
- *  Author: Andrew Tridgell, January 2013
- *
- */
 #pragma once
 
+
+#define __FILE FILE
+#include <functional>
+#undef __FILE
+
 #include <AP_Param/AP_Param.h>
 #include <AP_HAL/Util.h>
 #include <AP_Math/AP_Math.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Vehicle/AP_Vehicle.h>
 
-#define AP_SCHEDULER_NAME_INITIALIZER(_name) .name = #_name,
-
-/*
-  useful macro for creating scheduler task table
- */
-#define SCHED_TASK_CLASS(classname, classptr, func, _rate_hz, _max_time_micros) { \
-    .function = FUNCTOR_BIND(classptr, &classname::func, void),\
-    AP_SCHEDULER_NAME_INITIALIZER(func)\
-    .rate_hz = _rate_hz,\
-    .max_time_micros = _max_time_micros\
-}
-
-/*
-  A task scheduler for APM main loops
 
-  Sketches should call scheduler.init() on startup, then call
-  scheduler.tick() at regular intervals (typically every 10ms).
+extern const AP_HAL::HAL& hal;
 
-  To run tasks use scheduler.run(), passing the amount of time that
-  the scheduler is allowed to use before it must return
- */
 
-#include <AP_HAL/AP_HAL.h>
-#include <AP_Vehicle/AP_Vehicle.h>
+struct AP_Task {   
+    const char*                     name;
+    std::function<void()>           function;
+    float                           rate_hz;
+    uint16_t                        max_time_micros;
+    uint32_t                        last_run_ticks;
+    AP_HAL::Util::perf_counter_t    perf_counter;   
+};
 
-class AP_Scheduler
+// We define this as a constexpr, because of concerns regarding RAM usage in PX4
```

Comment:
Did you check usage before and after these changes?

Did this move stuff from .text into .data?